src/lib/APS2.cpp

```diff
@@ -487,58 +528,98 @@ void APS2::write_memory(const uint32_t & addr, const vector<uint32_t> & data){
 	 * data = vector<uint32_t> data
 	 */
 
-	if (data.size() == 0) {
-		return;
-	}
```

Comment:
I just checked and the unit tests do not currently include the empty message case. We should fix that and then revisit this.